### DISCUSSION TITLE
Follow the How to Install, it can't install on ubuntu24.04

### DISCUSSION THREAD

**Discussion by ltbyun** [louislam/dockge#652] [Ask for Help]
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### Description

no configuration file provided: not found

### 👟 Reproduction steps
```shell
# Create directories that store your stacks and stores Dockge's stack
sudo mkdir -p /opt/stacks /opt/dockge
cd /opt/dockge

# Download the compose.yaml
sudo curl https://raw.githubusercontent.com/louislam/dockge/master/compose.yaml --output compose.yaml

# Start the server
sudo docker compose up -d

# If you are using docker-compose V1 or Podman
# docker-compose up -d
```

### 👀 Expected behavior

docker compose up -d works

### 😓 Actual Behavior

it can't find the config file

### Dockge Version

latest

### 💻 Operating System and Arch

Ubuntu24.04 x64

### 🌐 Browser

Edge

### 🐋 Docker Version

24.0.5

**Comment by louislam** [comment]
Check if `compose.yaml` is really in the directory.

**Comment by Noxira** [comment]
Hello I am met with this same error running Ubuntu 22.04.

Apparently, it's because of docker is installed via snap.

<details><summary>File not found?</summary>
<p>

go into the directory

Running `pwd` results in `/opt/dockge`

Running `ls -l` results in 
```
total 4
-rw-r--r-- 1 root root 760 Apr  1 06:20 docker-compose.yml
```

But running `docker compose -f docker-compose.yml up -d` in that directory results in
`open /var/lib/snapd/void/docker-compose.yml: no such file or directory `

Apparently snap is doing something to safeguard access to some directories, hence not seeing the compose file

</p>
</details> 

This old [askubuntu question](https://askubuntu.com/a/1033617) mentioned that snap hardcodes /home for access, tried it in the user's home and yeah it worked just fine!

I'm now using the "Advanced" part of the [script](https://github.com/louislam/dockge?tab=readme-ov-file#advanced) for now

**Comment by barbalex** [comment]
I have exact same problem, right after freshly installing ubuntu 24 lts server.

It's a bit of a hassle for not so experienced users to follow basic instructions yet error, then not to be sure how exactly to change things to get it to work in a different path.

**Comment by t984447** [comment]
Nope, my comment is NOT related to above issue. My issue was a change in Podman which do not search docker.io for images unless docker.io is added to _/etc/containers/registries.conf_.
Added  _unqualified-search-registries = ["docker.io"]_
https://unix.stackexchange.com/questions/701784/podman-no-longer-searches-dockerhub-error-short-name-did-not-resolve-to-an



--------------------------------------
OLD ISSUE

I am not sure but I might have the same issue. 

- Ubuntu 24.04
- installed:  Podman, Podman-compose, docker-compose

And when trying to run _docker-compose up -d_ I get the following error in the terminal.
Not sure about the reference to _No module named 'distutils'_ but searching it says that distutils is no longer included in Python 3.12.
I have version _Python 3.12.3_


please note that I have tried with and without sudo.

> /opt/dockage/$ sudo docker-compose up -d
> Traceback (most recent call last):
>   File "/usr/bin/docker-compose", line 33, in <module>
>     sys.exit(load_entry_point('docker-compose==1.29.2', 'console_scripts', 'docker-compose')())
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/bin/docker-compose", line 25, in importlib_load_entry_point
>     return next(matches).load()
>            ^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/importlib/metadata/__init__.py", line 205, in load
>     module = import_module(match.group('module'))
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/importlib/__init__.py", line 90, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 995, in exec_module
>   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
>   File "/usr/lib/python3/dist-packages/compose/cli/main.py", line 9, in <module>
>     from distutils.spawn import find_executable
> ModuleNotFoundError: No module named 'distutils'



oddly enough I get another error with _podman-compose up -d_.

> /opt/dockge$ podman-compose up -d 
> podman-compose version: 1.0.6
> ['podman', '--version', '']
> using podman version: 4.9.3
> ** excluding:  set()
> ['podman', 'ps', '--filter', 'label=io.podman.compose.project=dockge', '-a', '--format', '{{ index .Labels "io.podman.compose.config-hash"}}']
> ['podman', 'network', 'exists', 'dockge_default']
> ['podman', 'network', 'create', '--label', 'io.podman.compose.project=dockge', '--label', 'com.docker.compose.project=dockge', 'dockge_default']
> ['podman', 'network', 'exists', 'dockge_default']
> podman run --name=dockge_dockge_1 -d --label io.podman.compose.config-hash=5e5fcc34973e37bf191d036d1a03311796a97c46f15c52c51aa3a58ee0c3fede --label io.podman.compose.project=dockge --label io.podman.compose.version=1.0.6 --label PODMAN_SYSTEMD_UNIT=podman-compose@dockge.service --label com.docker.compose.project=dockge --label com.docker.compose.project.working_dir=/opt/dockge --label com.docker.compose.project.config_files=compose.yaml --label com.docker.compose.container-number=1 --label com.docker.compose.service=dockge -e DOCKGE_STACKS_DIR=/opt/stacks -v /var/run/docker.sock:/var/run/docker.sock -v /opt/dockge/data:/app/data -v /opt/stacks:/opt/stacks --net dockge_default --network-alias dockge -p 5001:5001 --restart unless-stopped louislam/dockge:1
> Error: short-name "louislam/dockge:1" did not resolve to an alias and no unqualified-search registries are defined in "/etc/containers/registries.conf"
> exit code: 125
> podman start dockge_dockge_1
> Error: no container with name or ID "dockge_dockge_1" found: no such container
> exit code: 125
